### DISCUSSION TITLE
Suggestion for default output format

### DISCUSSION THREAD

**Discussion by DannyBen** [rubocop/rubocop#11497] [Ideas]
To me, the default output of rubocop always seems messy.

- Things that are the same (Correctable) are highlighted
- Things that are different (cop name) are not highlighted, and not aligned
- Things that I care about last (path) appear first.

I always find myself creating the todo file so I can better understand the output.

So I wonder if there is openness to consider slight adjustments, and if not, I wonder how difficult it would be for me to create a custom formatter that is always used.

I took the liberty to do a rough mock.

### Standard rubocop output

(also keep in mind, when the terminal is narrower, word wrapping makes it even less readable)
![image](https://user-images.githubusercontent.com/2405099/214402960-fa65da23-a2a0-409b-9309-ab48a07bfd0d.png)

### Modified output

First, the cop, then the message, then the location.
![image](https://user-images.githubusercontent.com/2405099/214403137-f493d281-561c-470b-8b17-741a4c9edd02.png)

**Comment by airled** [comment]
The [documentation](https://docs.rubocop.org/rubocop/extensions.html#custom-formatters) suggests that you can create your own formatters by subclassing `RuboCop::Formatter::BaseFormatter` class and then use it in your cli or config file.

So you can do these steps. Create file `myformatter.rb`. For example, its content is
```ruby
class MyFormatter < RuboCop::Formatter::BaseFormatter
  def file_finished(file, offenses)
    return if offenses.empty?

    offenses.each do |offense|
      output.puts("#{offense.message}\n in #{file}")
    end
  end
end
```

And now you can use it as formatter
`rubocop --require ./myformatter.rb --format MyFormatter myfile.rb`

And the output will have such style:
```
Style/FrozenStringLiteralComment: Missing frozen string literal comment.
 in /home/air/myfile.rb
Lint/ShadowingOuterLocalVariable: Shadowing outer local variable - `userfile`.
 in /home/air/myfile.rb
```

You can add colors and other information for your purposes in your formatter class.

**Reply by DannyBen** [reply]
Thank you, this is helpful.
I was hoping to demonstrate that the default rubocop output can be improved for the sake of all users, not just me.
